### ISSUE TITLE
Prevent interactions stealing the category-specific intercept variance

### ISSUE THREAD

**Issue by athowes** [athowes/multi-agyw#42]
I used to not include category random effects in the model, but then I thought that it would be strange not to have first order effects if I'm including interactions. Anyway, it turns out that the category effects are basically not doing anything. This is a bit of an issue because I think there is sufficient flexibility in any of the interaction random effects to subsume the category random effects. And since we care about the proportion variance this is bad, since it seems somewhat arbitrary which of the interaction effects actually takes the category part. I think the solution is some kind of constraint on all of the interaction random effects to stop them absorbing the category random effects. I'm already constraining them to sum-to-zero as in [here](https://github.com/athowes/multi-agyw/blob/cc7938cce0ea6381b16d0c48b700543d675d652e/src/aaa_fit_all-dhs-multi-sexbehav-sae/script.R#L272) so I think it'll have to be something more complicated. Possibly constrain to sum to zero within each `i`, `t` or `a`, not only over all `ik`, `tk` or `ak`.

In other words, say for the age-category random effects, what is required is:

`sum_a \alpha_{ak} = 0 \forall k = 1, ..., K`

whereas what is there at the moment is

`sum_{k = 1}^K \sum_a \alpha_{ak} = 0`
